Diagrams

Tylores opened the pull request:
This has become a gigantic blob of changes that must be addressed before it grows.

Tylores committed:
still need to finish up the waterheater systems using a target temps and gallon usage.

Tylores committed:
System stubs done for water heater simulator

Tylores committed:
nearly done with the updated demo. Need to add the metering resource and then that will be most of it.

Tylores committed:
Updated the remaining resources we will be using for the trust demo. Updated the trust xml to include timestamp and changed status code element name.

Tylores committed:
testing the waters with mermaid.js

Tylores committed:
Update README.md

Tylores committed:
Update README.md

Tylores committed:
Update README.md

Tylores committed:
Update README.md

Tylores committed:
Update README.md

Tylores committed:
Update README.md

Tylores committed:
still testing out some of the new figures.

Tylores committed:
Merge branch 'diagrams' of https://github.com/PortlandStatePowerLab/doe-egot-system into diagrams

Tylores committed:
restructure and clean up of DCM along with sep modifications

Tylores committed:
continued renaming sep objects for ecs and better std alignment

Tylores committed:
complete update of sep objects and typing

Tylores committed:
updates to xml adapters for modified sep objects

Tylores committed:
towards fixing xml adapters and tests with new sep objects

Tylores committed:
xml adapters and tests are working as expected and the ecs systems are ready to be implemented using the new objects

Tylores committed:
clients and server now use the updated sep objects and demo is working correctly.

Tylores committed:
reorganization of https server code as there cannot be a generic server. Added base ecs module code for server.

Tylores committed:
update ecs modules, will be reconfiguring the access module to use the child of feature in flecs

Tylores committed:
working towards the updated http methods with the gsp

Tylores committed:
still working towards the updates for the gsp server using new sep objects.

Tylores committed:
updated cmake instructions for a debug/release build and fixed all build errors. Now further development should go much smoother

Tylores committed:
switching back from tablet

Tylores committed:
There is something weird going on with cmake and I think it has to do with some naming. I am going through each of the subdirectories and cleaning up names and header guards.

Tylores committed:
Updated cmakefiles to use more modern syntax. Cleaned up file structure to simplify building and modifying.

Tylores committed:
Still cleaning things up and moving things around. The dtm no shall just be ran with python as the boost python wrapper only  seemed to complicate things. The dcm is now building correctly using new structure.

Tylores committed:
cmake updates are done and everything is building and linking correctly.

MidrarAdham reviewed (changes requested):
@Tylores There is no setup.sh file in the doe-got-system directory.

Tylores committed:
The server/client interactions don't appear to be working correclty. I think I am just not initializing the entities correctly or there is a scoping issue with the requests.

Tylores committed:
optionals are what we were missing. I need to update the adapters to not load data that hasn't been initialized.

Tylores committed:
alright rather than rewriting the entire sep models and adapters I am going to check for empty hrefs for each element and skip mapping if blank. That seems to work great and won't require a ton of extra work.

Tylores committed:
ok after some tinkering I think boost optional will be required. We need to know when a value has actually been initialized or if it has just been zero initialized. The only way to do this is throught boost optional

Tylores committed:
working towards updating all models with boost optional.

Tylores committed:
boost optional is taking a while, but there are just a large number in dcap and edev. The rest should go much faster

Tylores committed:
the tests have been updated to use boost optional and are nearly finished. Still getting some issues, but those should be worked out soon.

Tylores committed:
ok now all the adapters and tests have been updated. I missed a few before. I am still getting optional errors which is strange as all values are being populated. I think it has to be in the adapter and I need to use a different method to set values for optionals.

Tylores committed:
alright boost optional adapter is working correctly. It isn't perfect, but its done for dcap. I will wrap up the rest of the adapters tomorrow.

Tylores committed:
The end device DeviceCategoryType is causing issues now for some reason. It isn't getting translated and I ralized casting doesn't work for a specific hex mask type data value. While the enum type will allow masking of a bit position it doesn't technically hold all the bits at once.

Tylores committed:
the hex values need to be read/written as a string. I had assumed I was doing an extra step before, but apparently this is something I figured out a long time ago and forgot.

Tylores committed:
For some reason now the changedTime element is causing the xml validation to fail. It is saying that is is not allowed for the ocntent model, but the sample xml includes it and it passes so I have no idea what is happening. I have triple checked the path and even removed the line and rewrote it to see if there was some hidden unicode char that was messing things up.

Tylores committed:
alright I think there is something weird happening since dcap is working find, but sdev and edev are having issues. Tracing down all the path variables to see what is happening

Tylores committed:
Online validators are also saying the adapter is invalid so there is some weird formatting thing that is happening because the adapter values are correct.

Tylores committed:
apparently the sequence declaration in xml schema means it has to appear in the exact order, but the error message just gives you an element not allowed error so that is frustrating

Tylores committed:
finally wrapped up the optional additions and tests are working again

Tylores committed:
addition of optional tests for primary components with a large set of optionals.

Tylores committed:
starting the cleanup of the client and server using optionals. Next move construction of entities out of the http methods within the world

Tylores committed:
alright I was using the lookup method wrong which was causing all kinds of issues. I need to ensure that lookup doesn't return an empty entity. Much like the boost optional work we need to be carefule when and where we access memory

Tylores committed:
alright there are alot of moving parts involved when changing the uri path checks. Things are working correctly, but you just need to remember to update wadl and uri files when a new route is added or updated.

Tylores committed:
adding routes to resources isn't the most intuitive as some are world resources and some are client specific and there isn't a way to distinquish at this time.

Tylores committed:
still a few world methods that need to be modified for the response resources.

Tylores committed:
responses are now working correctly. The last thing is creating a power status test since that one is missing

Tylores committed:
power status up and running. Now I just need to finalize the gsp and dcm systems and we will be up and running

Tylores committed:
starting pulling over the systems for the dcm.

Tylores committed:
alright apparently cascading module imports doesn't work as expected, it isn't too hard to import all modules within main so thats good.

Tylores committed:
I found that you can create an observer with multiple components. However they only look at entities that have both. I think I will need to make and observer for FRQs and then do a world query to find the link list to post to from whichever module is implementing the request. This should make it pretty easy for SunSpec Modbuse, CTA 2045, etc

Tylores committed:
updated the ssl gen script to capture registration info for dcm and gsp

Tylores committed:
currently I am going to keep the waterdraw profiles for now

Tylores committed:
ecs singleton clock working correctly now

Tylores committed:
the systems that us singletons don't seem to be getting called correctly.

Tylores committed:
There was a weird clock sync issue between the client and server. I expect it was something to do with int overflow, but I think restructuring how the sync happens fixed the issue. Now there is only a one second offest here and there.

Tylores committed:
added a script for running the egot system and updated readme.

sonalinf committed:
For ease of instllation I moved the setup.sh file to the main folder doe-egot-system

I have relocated the 'setup.sh' file to the main folder 'doe-egot-system.' This adjustment facilitates the smooth flow of instructions.

sonalinf committed:
Merge branch 'diagrams' of https://github.com/PortlandStatePowerLab/doe-egot-system into diagrams

sonalinf committed:
Moved the setup.sh to tools folder for organization purposes.

Tylores committed:
temp shifting work for my laptop

Tylores committed:
still working on the waterheater systems

Tylores committed:
schedule updated is finally working correctly. Next waterheater event system and temperature system.

Tylores committed:
The waterheater systems are nearly there. The state isn't working and I cam trying to figure out why one system is blocking another

Tylores committed:
I was begin stupid and didn't give each of the water heater systems a unique name. That was the issue and I wish there was a better thing for that. Maybe leaving the name part empty would be a safer option since we don't use the name anyways? idk

Tylores committed:
linking the sep model to the water heater. Next link the response to the shed system to know when to end shed

Tylores committed:
alright now the energy calculation looks better. Need to double check the old paper to see where the upper and lower bounds should actually be for each of the modes. The FR response field for the system needs to be optional since it may not get one and if it doesn't then the system can just assume its request was ignored after a period of time.

Tylores committed:
water heater sim is in a good place. Revamping the flow reservation systems currently, but other systems can run without that currently.

Tylores committed:
Update setup.sh

xerces broke link to source files again

Tylores committed:
adding a barebones ecs to test the raspberry pi

Tylores committed:
Update setup.sh

Tylores committed:
update to flecs master for now until a new version is release with the raspberry pi fix

Tylores committed:
time sync format for gsp

Tylores committed:
simple update to allow flecs explorer app to visualize server

Tylores committed:
updated registration ssl to capture areas

Tylores committed:
updated ME resource files and gsp registration process

Tylores committed:
removing flecs explorer for now as its warnings are annoying

Tylores committed:
simple example of the service resource distribution between parent and children

Tylores committed:
added doxygen build instruction for documentation and final testing of gsp topology systems and service test

Tylores committed:
updating comments for doxygen, mostly lib/sep

Tylores committed:
cleaning doxygen generated folders that shouldn't exist

Tylores committed:
updated code documentation which caused me to change some code ordering that some how broke the server tests. Currently trying to track down what is happening now.

Tylores committed:
there was some funky behavior with the ECS Module constructors and pregeneration behavior so those functions were pulled from member functions. There was also a registration delay that was causing tests to fail because they would be ran during the start of the World registration.

Tylores committed:
updated cta2045 interface for ucm/device and added dcm-aosmith in apps

Tylores committed:
updated trust library for cta2045 device client

Tylores committed:
updated dcm-aosmith to use the trust cta2045 device

Tylores committed:
updated the dcm-aosmith to use the simple ecs with a commodity system

Tylores committed:
trust::cta2045 client was posting to the wront dtm path for messages

Tylores committed:
refactored the trust ucm to use commodity enums and added observers for commodity ecs

Tylores committed:
cannot init an enum class with an int

Tylores committed:
figuring out why the commodity read sucks

Tylores committed:
potential fix to prevent flecs and cea2045 libs from being downloaded and compiled after every git pull

Tylores committed:
ignore stream truncated error in https client

Tylores committed:
fixing cta2045 commodity read method pointer issue

Tylores committed:
hopefully fixed the fetch content download issue every build

Tylores committed:
fixed cta2045 observers to actually trigger on set

Tylores committed:
fixing the serial port path for the aosmith

Tylores committed:
addition of a cta2045 comms test for the aosmith dcm

MidrarAdham committed:
the commdity read and xml weird symbol are fixed. added opcode and comm status to solve the AppNak for the requests

Tylores committed:
updating sep xml adapters and tests for registration process

Tylores committed:
nearly done with the der settings update

Tylores committed:
still getting weird multiplier read issues for some reason.

Tylores committed:
the updated der adapters are complete and tests are mostly running as expected. I think there is something wierd with the schema or xerces, but the translation accuracy is correct so that should be fine

Tylores committed:
updated ecs client/server for der resources
